<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="SarahL." />


<title>Subcluster Annotation</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/yeti.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">RA SingleCellAnalysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="about.html">
    <span class="fa fa-info"></span>
     
    About
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-bar-chart"></span>
     
    Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Preprocessing</li>
    <li>
      <a href="00_DataPreparation.html">Data Preparation</a>
    </li>
    <li>
      <a href="01_QualityControl.html">Quality Control</a>
    </li>
    <li>
      <a href="02_DimensionalityReduction.html">Dimensionality Reduction</a>
    </li>
    <li>
      <a href="03_Integration_Batchelor.html">Integration</a>
    </li>
    <li>
      <a href="04_CellCycle.html">Cellcycle Assignement</a>
    </li>
    <li>
      <a href="05_Clustering.html">Clustering</a>
    </li>
    <li>
      <a href="06_CelltypeAnnotation.html">Annotation</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">Subclustering</li>
    <li>
      <a href="07_Subclustering.html">Create Subclustering</a>
    </li>
    <li>
      <a href="07_SubClustering_Annotation.html">Subclustering Annotation</a>
    </li>
    <li>
      <a href="07_SubClustering_Merge.html">Merge Subclusters</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">Analysis</li>
    <li>
      <a href="08_DifferentialAbundance.html">Differential Abundance</a>
    </li>
    <li>
      <a href="09_SC_Fibroblast.html">Fibroblast Subcluster Profiling</a>
    </li>
    <li>
      <a href="09_SC_Macrophage.html">Macrophage Subcluster Profiling</a>
    </li>
    <li>
      <a href="10_DifferentialExpression_MAST.html">Differential Expression</a>
    </li>
    <li>
      <a href="11_EnrichmentAnalysis.html">Enrichment Analysis</a>
    </li>
    <li>
      <a href="11_EnrichmentAnalysis.html">Cell Cell Communication</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">Spatial Transcriptomics</li>
    <li>
      <a href="Spatial_knee.html">Spatial Transcriptomics Knee</a>
    </li>
    <li>
      <a href="Spatial_hand.html">Spatial Transcriptomics Hand</a>
    </li>
  </ul>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sarloet/RA_SingleCellAnalysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Subcluster Annotation</h1>
<h3 class="subtitle">07_SubClustering_Merge</h3>
<h4 class="author">SarahL.</h4>
<h4 class="date">Jun 27 2024 22:31:52</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-06-27
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>RA_SingleCellAnalysis/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240328code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240328)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240328code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240328)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsarloetRASingleCellAnalysistree5887ab5acd04bce3d1a3c048b1b47c5817fd9520targetblank5887ab5a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/sarloet/RA_SingleCellAnalysis/tree/5887ab5acd04bce3d1a3c048b1b47c5817fd9520" target="_blank">5887ab5</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsarloetRASingleCellAnalysistree5887ab5acd04bce3d1a3c048b1b47c5817fd9520targetblank5887ab5a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/sarloet/RA_SingleCellAnalysis/tree/5887ab5acd04bce3d1a3c048b1b47c5817fd9520" target="_blank">5887ab5</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/figure/
    Ignored:    data/.DS_Store
    Ignored:    data/cellbender_data_h5/
    Ignored:    data/spatial/
    Ignored:    output/00_DoubletDetection.rds
    Ignored:    output/00_sce_DataPreparation.rds
    Ignored:    output/01_sce_QualityControl.rds
    Ignored:    output/01_sce_QualityControl_cellfiltered.rds
    Ignored:    output/02_sce_DimensionalityReduction.rds
    Ignored:    output/03_sce_Integration_Batchelor.rds
    Ignored:    output/03_sce_Integration_Harmony.rds
    Ignored:    output/04_sce_CellCycle.rds
    Ignored:    output/05_sce_CelltypeAnnotation.rds
    Ignored:    output/05_sce_Clustering.rds
    Ignored:    output/06_sce_CelltypeAnnotation.rds
    Ignored:    output/07_sce_SC_Bcell.rds
    Ignored:    output/07_sce_SC_Dendritic.rds
    Ignored:    output/07_sce_SC_Dendritic_Anno.rds
    Ignored:    output/07_sce_SC_Endothelial.rds
    Ignored:    output/07_sce_SC_Endothelial_Anno.rds
    Ignored:    output/07_sce_SC_Fibroblast.rds
    Ignored:    output/07_sce_SC_Fibroblast_Anno.rds
    Ignored:    output/07_sce_SC_Macrophage.rds
    Ignored:    output/07_sce_SC_Macrophage_Anno.rds
    Ignored:    output/07_sce_SC_Mast.rds
    Ignored:    output/07_sce_SC_Merge.rds
    Ignored:    output/07_sce_SC_Neutrophil.rds
    Ignored:    output/07_sce_SC_Plasma.rds
    Ignored:    output/07_sce_SC_SmoothMuscle.rds
    Ignored:    output/07_sce_SC_Tcell.rds
    Ignored:    output/07_sce_SC_Tcell_Anno.rds
    Ignored:    output/08_DifferentialAbundance_miloR.rds
    Ignored:    output/10_sce_CellChat_Knee.rds
    Ignored:    output/10_sce_CellChat_MCP.rds
    Ignored:    output/10_sce_CellChat_Wrist.rds
    Ignored:    output/Annotation_L0_Wilcoxon.xlsx
    Ignored:    output/Annotation_L1_Wilcoxon.xlsx
    Ignored:    output/Annotation_output/
    Ignored:    output/Cluster_Marker_Genes.xlsx
    Ignored:    output/Knee_KvM.csv
    Ignored:    output/Knee_KvW.csv
    Ignored:    output/MAST_output/
    Ignored:    output/MCP_KvM.csv
    Ignored:    output/MCP_MvW.csv
    Ignored:    output/ManualAnno_L0_Wilcoxon.xlsx
    Ignored:    output/ManualAnno_L1_Wilcoxon.xlsx
    Ignored:    output/ScMergeAnnotation_output/
    Ignored:    output/Spatial_knee.rds
    Ignored:    output/Spatial_knee_SC.rds
    Ignored:    output/Subcluster_Annotation_output/
    Ignored:    output/Wrist_KvW.csv
    Ignored:    output/Wrist_MvW.csv
    Ignored:    renv/.DS_Store
    Ignored:    renv/library/
    Ignored:    renv/staging/

Untracked files:
    Untracked:  analysis/07_SubClustering_Merge.Rmd
    Untracked:  analysis/08_DifferentialExpression_EdgeR.Rmd
    Untracked:  analysis/08_DifferentialExpression_MAST.Rmd
    Untracked:  analysis/08_DifferentialExpression_Pseudobulk.Rmd
    Untracked:  analysis/08_DifferentialExpression_muscat.Rmd
    Untracked:  analysis/09_DifferentialAbundance_miloR.Rmd
    Untracked:  analysis/10_EnrichmentAnalysis.Rmd
    Untracked:  analysis/11_CellChat.Rmd
    Untracked:  analysis/99_Plotting.Rmd
    Untracked:  analysis/Spatial_hand.Rmd
    Untracked:  analysis/Spatial_knee.Rmd

Unstaged changes:
    Modified:   analysis/02_DimensionalityReduction.Rmd
    Modified:   analysis/03_Integration_Batchelor.Rmd
    Modified:   analysis/06_CelltypeAnnotation.Rmd
    Modified:   analysis/06_SC_Fibroblast.Rmd
    Deleted:    analysis/07_DifferentialExpression_EdgeR.Rmd
    Deleted:    analysis/07_DifferentialExpression_MAST.Rmd
    Deleted:    analysis/07_DifferentialExpression_Pseudobulk.Rmd
    Deleted:    analysis/07_DifferentialExpression_muscat.Rmd
    Modified:   analysis/07_SubClustering.Rmd
    Modified:   analysis/07_SubClustering_Annotation.Rmd
    Deleted:    analysis/08_DifferentialAbundance_miloR.Rmd
    Deleted:    analysis/09_EnrichmentAnalysis.Rmd
    Deleted:    analysis/10_Plotting.Rmd
    Deleted:    analysis/Ege_06_SC_Fibroblast.Rmd
    Deleted:    analysis/OLD06_SubClustering.Rmd
    Modified:   analysis/_site.yml
    Modified:   analysis/index.Rmd
    Modified:   code/standard_libraries.R
    Modified:   renv.lock

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="subcluster-annotation" class="section level1">
<h1>Subcluster Annotation</h1>
<div id="setup" class="section level2">
<h2>Setup</h2>
<div id="standard-packages" class="section level3">
<h3>Standard packages</h3>
<pre class="r"><code>library(here)
source(here(&quot;code&quot;, &quot;standard_libraries.R&quot;))</code></pre>
</div>
<div id="additional-packages" class="section level3">
<h3>Additional Packages</h3>
<pre class="r"><code>#Load Packages

suppressPackageStartupMessages({
library(intrinsicDimension)
library(patchwork)
library(presto)
library(xlsx)
})</code></pre>
</div>
<div id="set-parameter" class="section level3">
<h3>Set Parameter</h3>
<pre class="r"><code>set.seed(100)
bpp &lt;- BiocParallel::MulticoreParam(parallel::detectCores()-1, RNGseed=100)
path &lt;- here::here()

#Output path
output_dir &lt;- here(&quot;output&quot;,&quot;ScMergeAnnotation_output&quot;)
if(!dir.exists(output_dir)) dir.create(output_dir)</code></pre>
</div>
<div id="load-unnannotated-sc-data" class="section level3">
<h3>Load unnannotated SC data</h3>
<pre class="r"><code>hvg_list&lt;-c()</code></pre>
<pre class="r"><code>#Load sce set Annotation_L2 to clustername &amp; remove hvg.sc
sce.SMC &lt;- readRDS(file = paste0(path,&#39;/output/07_sce_SC_SmoothMuscle.rds&#39;))
sce.SMC$Annotation_L2 &lt;- &quot;Smooth muscle cell&quot;
hvg_list&lt;-union(hvg_list, rownames(sce.SMC)[rowData(sce.SMC)$hvg.sc])
rowData(sce.SMC)$hvg.sc &lt;- NULL

sce.Mast &lt;- readRDS(file = paste0(path,&#39;/output/07_sce_SC_Mast.rds&#39;))
sce.Mast$Annotation_L2 &lt;- &quot;Mast cell&quot;
hvg_list&lt;-union(hvg_list, rownames(sce.Mast)[rowData(sce.Mast)$hvg.sc])
rowData(sce.Mast)$hvg.sc &lt;- NULL

sce.Bcell &lt;- readRDS(file = paste0(path,&#39;/output/07_sce_SC_Bcell.rds&#39;))
sce.Bcell$Annotation_L2 &lt;- &quot;B cell&quot;
hvg_list&lt;-union(hvg_list, rownames(sce.Bcell)[rowData(sce.Bcell)$hvg.sc])
rowData(sce.Bcell)$hvg.sc &lt;- NULL

sce.Plasma &lt;- readRDS(file = paste0(path,&#39;/output/07_sce_SC_Plasma.rds&#39;))
sce.Plasma$Annotation_L2 &lt;- &quot;Plasma&quot;
hvg_list&lt;-union(hvg_list, rownames(sce.Plasma)[rowData(sce.Plasma)$hvg.sc])
rowData(sce.Plasma)$hvg.sc &lt;- NULL

sce.Neutrophil &lt;- readRDS(file = paste0(path,&#39;/output/07_sce_SC_Neutrophil.rds&#39;))
sce.Neutrophil$Annotation_L2 &lt;- &quot;Neutrophil&quot;
hvg_list&lt;-union(hvg_list, rownames(sce.Neutrophil)[rowData(sce.Neutrophil)$hvg.sc])
rowData(sce.Neutrophil)$hvg.sc &lt;- NULL</code></pre>
</div>
<div id="load-annotated-sc-data" class="section level3">
<h3>Load annotated SC data</h3>
<pre class="r"><code>#Load sce &amp; remove is_hvg_sc
sce.Fibroblast &lt;- readRDS(file = paste0(path,&#39;/output/07_sce_SC_Fibroblast_Anno.rds&#39;))
hvg_list&lt;-union(hvg_list, rownames(sce.Fibroblast)[rowData(sce.Fibroblast)$hvg.sc])
rowData(sce.Fibroblast)$hvg.sc &lt;- NULL

sce.Macrophage &lt;- readRDS(file = paste0(path,&#39;/output/07_sce_SC_Macrophage_Anno.rds&#39;))
hvg_list&lt;-union(hvg_list, rownames(sce.Macrophage)[rowData(sce.Macrophage)$hvg.sc])
rowData(sce.Macrophage)$hvg.sc &lt;- NULL

sce.Tcell &lt;- readRDS(file = paste0(path,&#39;/output/07_sce_SC_Tcell_Anno.rds&#39;))
hvg_list&lt;-union(hvg_list, rownames(sce.Tcell)[rowData(sce.Tcell)$hvg.sc])
rowData(sce.Tcell)$hvg.sc &lt;- NULL

sce.Endothelial &lt;- readRDS(file = paste0(path,&#39;/output/07_sce_SC_Endothelial_Anno.rds&#39;))
hvg_list&lt;-union(hvg_list, rownames(sce.Endothelial)[rowData(sce.Endothelial)$hvg.sc])
rowData(sce.Endothelial)$hvg.sc &lt;- NULL

sce.Dendritic &lt;- readRDS(file = paste0(path,&#39;/output/07_sce_SC_Dendritic_Anno.rds&#39;))
hvg_list&lt;-union(hvg_list, rownames(sce.Dendritic)[rowData(sce.Dendritic)$hvg.sc])
rowData(sce.Dendritic)$hvg.sc &lt;- NULL</code></pre>
</div>
<div id="merge-subclusters" class="section level3">
<h3>Merge Subclusters</h3>
<pre class="r"><code>#Indicate discarted cells
#colData(sce)$subcluster &lt;- &quot;discard&quot;
sce.merged &lt;- cbind(sce.SMC,sce.Mast,sce.Bcell,sce.Plasma,sce.Neutrophil,sce.Fibroblast, sce.Macrophage,sce.Tcell,sce.Endothelial,sce.Dendritic)
#add hvgs for subclustering
rowData(sce.merged)$is_hvg_sc &lt;- rownames(sce.merged) %in% hvg_list

rm(sce.SMC,sce.Mast,sce.Bcell,sce.Plasma,sce.Neutrophil,sce.Fibroblast, sce.Macrophage,sce.Tcell,sce.Endothelial,sce.Dendritic,hvg_list)
gc()</code></pre>
<pre><code>            used   (Mb) gc trigger    (Mb) limit (Mb)   max used    (Mb)
Ncells  10096903  539.3   14945550   798.2         NA   14641526   782.0
Vcells 733350778 5595.1 2251402099 17176.9     102400 2739756837 20902.7</code></pre>
</div>
</div>
<div id="realculate-pca-and-umap" class="section level2">
<h2>Realculate PCA and UMAP</h2>
<pre class="r"><code>#Get top HVG for subcluster
dec &lt;- modelGeneVar(sce.merged, block=sce.merged$Sample, density.weights=FALSE,BPPARAM=bpp)
top.hvgs &lt;- getTopHVGs(dec, n=3000)
rowData(sce.merged)$is_hvg_merge &lt;- rownames(sce.merged) %in% top.hvgs


#Get recalculated PCA on Merge
sce.merged&lt;-runPCA(sce.merged,exprs_values =&quot;reconstructed&quot;, subset_row = rowData(sce.merged)$is_hvg_merge, name = &quot;MNN_merge&quot;)

#Get PC elbow
elbow &lt;- ceiling(as.numeric(maxLikGlobalDimEst(as.matrix(reducedDim(sce.merged, &quot;MNN_merge&quot;)), k=5)))
cat(paste0(&quot;Elbow Point: &quot;,elbow))</code></pre>
<pre><code>Elbow Point: 17</code></pre>
<pre class="r"><code>reducedDim(sce.merged, &#39;MNN_merge_reduced&#39;) &lt;- reducedDim(sce.merged, &#39;MNN_merge&#39;)[,seq_len(elbow)]
  
#Get recalculated UMAP on Merge
sce.merged &lt;- runUMAP(sce.merged,name = &quot;MNN_merge_reduced&quot;, dimred = &#39;MNN_merge_reduced&#39;,subset_row=rowData(sce.merged)$is_hvg_merge)</code></pre>
<div id="plot-umap" class="section level3 tabset">
<h3 class="tabset">Plot UMAP</h3>
<div id="annotation-level-2" class="section level4">
<h4>Annotation Level 2</h4>
<pre class="r"><code>plotReducedDim(sce.merged, dimred=&quot;MNN_merge_reduced&quot;, colour_by=&quot;Annotation_L2&quot;,point_alpha=0.7,point_size=0.1)+
  labs( x=&#39;UMAP 1&#39;, y=&#39;UMAP 2&#39; )+
  guides(colour = guide_legend(override.aes = list(size=2)))</code></pre>
<p><img src="figure/07_SubClustering_Merge.Rmd/UMAPL2-1.png" width="100%" height="100%" style="display: block; margin: auto;" /></p>
</div>
<div id="annotation-level-1" class="section level4">
<h4>Annotation Level 1</h4>
<pre class="r"><code>plotReducedDim(sce.merged, dimred=&quot;MNN_merge_reduced&quot;, colour_by=&quot;Annotation_L1&quot;,point_alpha=0.7,point_size=0.1)+
  labs( x=&#39;UMAP 1&#39;, y=&#39;UMAP 2&#39; )+
  guides(colour = guide_legend(override.aes = list(size=2)))</code></pre>
<p><img src="figure/07_SubClustering_Merge.Rmd/UMAPL1-1.png" width="100%" height="100%" style="display: block; margin: auto;" /></p>
</div>
<div id="annotation-level-0" class="section level4">
<h4>Annotation Level 0</h4>
<pre class="r"><code>plotReducedDim(sce.merged, dimred=&quot;MNN_merge_reduced&quot;, colour_by=&quot;Annotation_L0&quot;,point_alpha=0.7,point_size=0.1)+
  labs( x=&#39;UMAP 1&#39;, y=&#39;UMAP 2&#39; )+
  guides(colour = guide_legend(override.aes = list(size=2)))</code></pre>
<p><img src="figure/07_SubClustering_Merge.Rmd/UMAPL0-1.png" width="100%" height="100%" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="plot-pca" class="section level3 tabset">
<h3 class="tabset">Plot PCA</h3>
<div id="annotation-level-2-1" class="section level4">
<h4>Annotation Level 2</h4>
<pre class="r"><code>plotReducedDim(sce.merged, dimred=&quot;MNN_merge&quot;, colour_by=&quot;Annotation_L2&quot;,point_alpha=0.7,point_size=0.1)+
  labs( x=&#39;UMAP 1&#39;, y=&#39;UMAP 2&#39; )+
  guides(colour = guide_legend(override.aes = list(size=2)))</code></pre>
<p><img src="figure/07_SubClustering_Merge.Rmd/PCAL2-1.png" width="100%" height="100%" style="display: block; margin: auto;" /></p>
</div>
<div id="annotation-level-1-1" class="section level4">
<h4>Annotation Level 1</h4>
<pre class="r"><code>plotReducedDim(sce.merged, dimred=&quot;MNN_merge&quot;, colour_by=&quot;Annotation_L1&quot;,point_alpha=0.7,point_size=0.1)+
  labs( x=&#39;UMAP 1&#39;, y=&#39;UMAP 2&#39; )+
  guides(colour = guide_legend(override.aes = list(size=2)))</code></pre>
<p><img src="figure/07_SubClustering_Merge.Rmd/PCAL1-1.png" width="100%" height="100%" style="display: block; margin: auto;" /></p>
</div>
<div id="annotation-level-0-1" class="section level4">
<h4>Annotation Level 0</h4>
<pre class="r"><code>plotReducedDim(sce.merged, dimred=&quot;MNN_merge&quot;, colour_by=&quot;Annotation_L0&quot;,point_alpha=0.7,point_size=0.1)+
  labs( x=&#39;UMAP 1&#39;, y=&#39;UMAP 2&#39; )+
  guides(colour = guide_legend(override.aes = list(size=2)))</code></pre>
<p><img src="figure/07_SubClustering_Merge.Rmd/PCAL0-1.png" width="100%" height="100%" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="save-the-dataset" class="section level2">
<h2>Save the dataset</h2>
<pre class="r"><code>saveRDS(sce.merged, file =paste0(path,&#39;/output/07_sce_SC_Merge.rds&#39;))</code></pre>
</div>
<div id="wilcoxon-rank-sum-test-on-annotation" class="section level2">
<h2>Wilcoxon rank-sum test on Annotation</h2>
<pre class="r"><code>for (labeling in c(&#39;Annotation_L0&#39;,&#39;Annotation_L1&#39;,&#39;Annotation_L2&#39;)){
  #Wilcoxon rank-sum test
  markers_wilcox &lt;- wilcoxauc(sce.merged, group_by=labeling, assay = &#39;logcounts&#39;)
  topDEgenes &lt;- top_markers(markers_wilcox, n = 100, auc_min = 0.5, pval_max = 0.05)# filter for significant (p&lt;0.05) and over-expressed (auc&gt;0.5)
  wb = createWorkbook()
  for (i in unique(markers_wilcox$group) ){
    #get genes for the Anno
    markers&lt;-topDEgenes[,paste0(i), drop=TRUE]
    #Create table
    tmp&lt;-markers_wilcox[markers_wilcox$feature %in% markers &amp; markers_wilcox$group == i,]
    #Save table in Exelsheet
    addDataFrame(tmp, sheet=createSheet(wb, paste0(i)), startColumn=1, row.names=FALSE,col.names = TRUE)
  }
  saveWorkbook(wb,paste0( output_dir , &quot;/&quot;,paste0(labeling),&quot;_Wilcoxon.xlsx&quot;))  
}</code></pre>
</div>
<div id="show-removed-subclusterss" class="section level2">
<h2>Show removed subclusterss</h2>
<div id="load-data-before-subclustering" class="section level3">
<h3>Load Data before Subclustering</h3>
<pre class="r"><code>## RA DATASET
sce &lt;- readRDS(file = paste0(path,&#39;/output/06_sce_CelltypeAnnotation.rds&#39;)) # 06</code></pre>
</div>
<div id="mark-removed-cells" class="section level3">
<h3>Mark removed cells</h3>
<pre class="r"><code># Get discarded cells
discard &lt;- !(colnames(sce) %in% colnames(sce.merged))
# Add this as a new column to colData of sce1
sce$discard &lt;- discard</code></pre>
<pre class="r"><code>#Give Stats
cat(&quot;NR of Cells Before Subcluster QC: &quot;, dim(sce)[2],&quot;\n&quot;,
    &quot;NR of Cells After Subcluster QC: &quot;, dim(sce.merged)[2],&quot;\n&quot;,
    &quot;NR of Cells Filtered out: &quot;, dim(sce)[2] - dim(sce.merged)[2],&quot;\n&quot;,
    &quot;Cells Filtered out: [%]&quot;, (dim(sce)[2] - dim(sce.merged)[2])/dim(sce)[2]*100,&quot;\n&quot;
    )</code></pre>
<pre><code>NR of Cells Before Subcluster QC:  64639 
 NR of Cells After Subcluster QC:  61242 
 NR of Cells Filtered out:  3397 
 Cells Filtered out: [%] 5.255341 </code></pre>
<pre class="r"><code>p1&lt;-plotColData(sce, x=&quot;discard&quot;, y=&quot;sum&quot;,other_fields=&quot;discard&quot;,colour_by = &quot;discard&quot;) + 
    facet_wrap(~discard, nrow=1, scales = &quot;free_x&quot;) + 
    scale_y_log10() + 
    ggtitle(&quot;Total count&quot;)+
    guides(colour=guide_legend(title=&quot;Discarded&quot;))+
    theme(axis.text.x = element_text(angle = 45,hjust=1), axis.ticks.x=element_blank(),strip.text.x = element_blank())

p2&lt;-plotColData(sce, x=&quot;discard&quot;, y=&quot;detected&quot;,other_fields=&quot;discard&quot;,colour_by = &quot;discard&quot;) + 
    facet_wrap(~discard, nrow=1, scales = &quot;free_x&quot;) + 
    scale_y_log10() + 
    ggtitle(&quot;Detected features&quot;)+
    guides(colour=guide_legend(title=&quot;Discarded&quot;))+
    theme(axis.text.x = element_text(angle = 45,hjust=1), axis.ticks.x=element_blank(),strip.text.x = element_blank())

p3&lt;-plotColData(sce, x=&quot;discard&quot;, y=&quot;subsets_Mito_percent&quot;,other_fields=&quot;discard&quot;,colour_by = &quot;discard&quot;) + 
    facet_wrap(~discard, nrow=1, scales = &quot;free_x&quot;) + 
    scale_y_log10() + 
    ggtitle(&quot;Mito percent&quot;)+
    guides(colour=guide_legend(title=&quot;Discarded&quot;))+
    theme(axis.text.x = element_text(angle = 45,hjust=1), axis.ticks.x=element_blank(),strip.text.x = element_blank())

p1+p2+p3</code></pre>
<p><img src="figure/07_SubClustering_Merge.Rmd/QcPlots-1.png" width="100%" height="100%" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotReducedDim(sce, dimred=&quot;MNN_UMAP_reduced&quot;, colour_by=&quot;discard&quot;,point_alpha=0.7,point_size=0.1)+
  labs( x=&#39;UMAP 1&#39;, y=&#39;UMAP 2&#39; )+
  guides(colour = guide_legend(override.aes = list(size=2)))</code></pre>
<p><img src="figure/07_SubClustering_Merge.Rmd/UMAPcopm-1.png" width="100%" height="100%" style="display: block; margin: auto;" /></p>
<pre class="r"><code>print(Plot_QC_dimred(sce,dim=&quot;MNN_UMAP_reduced&quot;))</code></pre>
<p><img src="figure/07_SubClustering_Merge.Rmd/UMAPOc-1.png" width="100%" height="100%" style="display: block; margin: auto;" /></p>
<pre><code>TableGrob (2 x 3) &quot;arrange&quot;: 6 grobs
  z     cells    name           grob
1 1 (1-1,1-1) arrange gtable[layout]
2 2 (1-1,2-2) arrange gtable[layout]
3 3 (1-1,3-3) arrange gtable[layout]
4 4 (2-2,1-1) arrange gtable[layout]
5 5 (2-2,2-2) arrange gtable[layout]
6 6 (2-2,3-3) arrange gtable[layout]</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.3 (2024-02-29)
Platform: x86_64-apple-darwin20 (64-bit)
Running under: macOS Sonoma 14.4.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Warsaw
tzcode source: internal

attached base packages:
[1] stats4    stats     graphics  grDevices datasets  utils     methods  
[8] base     

other attached packages:
 [1] xlsx_0.6.5                  presto_1.0.0               
 [3] data.table_1.15.4           Rcpp_1.0.12                
 [5] patchwork_1.2.0             intrinsicDimension_1.2.0   
 [7] yaImpute_1.0-34             tidyr_1.3.1                
 [9] org.Hs.eg.db_3.18.0         AnnotationDbi_1.64.1       
[11] clusterProfiler_4.10.1      viridis_0.6.5              
[13] viridisLite_0.4.2           gridExtra_2.3              
[15] scran_1.30.2                scater_1.30.1              
[17] scuttle_1.12.0              SingleCellExperiment_1.24.0
[19] SummarizedExperiment_1.32.0 Biobase_2.62.0             
[21] GenomicRanges_1.54.1        GenomeInfoDb_1.38.8        
[23] IRanges_2.36.0              S4Vectors_0.40.2           
[25] BiocGenerics_0.48.1         MatrixGenerics_1.14.0      
[27] matrixStats_1.3.0           dplyr_1.1.4                
[29] ggplot2_3.5.1               BiocParallel_1.36.0        
[31] here_1.0.1                  workflowr_1.7.1            

loaded via a namespace (and not attached):
  [1] RcppAnnoy_0.0.22          splines_4.3.3            
  [3] later_1.3.2               bitops_1.0-7             
  [5] ggplotify_0.1.2           tibble_3.2.1             
  [7] polyclip_1.10-6           lifecycle_1.0.4          
  [9] edgeR_4.0.16              rprojroot_2.0.4          
 [11] processx_3.8.4            lattice_0.22-6           
 [13] MASS_7.3-60.0.1           magrittr_2.0.3           
 [15] limma_3.58.1              sass_0.4.9               
 [17] rmarkdown_2.27            jquerylib_0.1.4          
 [19] yaml_2.3.8                metapod_1.10.1           
 [21] httpuv_1.6.15             cowplot_1.1.3            
 [23] DBI_1.2.3                 RColorBrewer_1.1-3       
 [25] abind_1.4-5               zlibbioc_1.48.2          
 [27] purrr_1.0.2               ggraph_2.2.1             
 [29] RCurl_1.98-1.14           yulab.utils_0.1.4        
 [31] xlsxjars_0.6.1            tweenr_2.0.3             
 [33] git2r_0.33.0              GenomeInfoDbData_1.2.11  
 [35] enrichplot_1.22.0         ggrepel_0.9.5            
 [37] irlba_2.3.5.1             tidytree_0.4.6           
 [39] dqrng_0.4.1               DelayedMatrixStats_1.24.0
 [41] codetools_0.2-20          DelayedArray_0.28.0      
 [43] DOSE_3.28.2               ggforce_0.4.2            
 [45] tidyselect_1.2.1          aplot_0.2.3              
 [47] farver_2.1.2              ScaledMatrix_1.10.0      
 [49] jsonlite_1.8.8            BiocNeighbors_1.20.2     
 [51] tidygraph_1.3.1           tools_4.3.3              
 [53] treeio_1.26.0             glue_1.7.0               
 [55] SparseArray_1.2.4         xfun_0.45                
 [57] qvalue_2.34.0             withr_3.0.0              
 [59] BiocManager_1.30.23       fastmap_1.2.0            
 [61] bluster_1.12.0            fansi_1.0.6              
 [63] callr_3.7.6               digest_0.6.36            
 [65] rsvd_1.0.5                R6_2.5.1                 
 [67] gridGraphics_0.5-1        colorspace_2.1-0         
 [69] GO.db_3.18.0              RSQLite_2.3.7            
 [71] utf8_1.2.4                generics_0.1.3           
 [73] renv_1.0.7                graphlayouts_1.1.1       
 [75] httr_1.4.7                S4Arrays_1.2.1           
 [77] scatterpie_0.2.3          whisker_0.4.1            
 [79] uwot_0.2.2                pkgconfig_2.0.3          
 [81] rJava_1.0-11              gtable_0.3.5             
 [83] blob_1.2.4                XVector_0.42.0           
 [85] shadowtext_0.1.3          htmltools_0.5.8.1        
 [87] fgsea_1.28.0              scales_1.3.0             
 [89] png_0.1-8                 ggfun_0.1.5              
 [91] knitr_1.47                rstudioapi_0.16.0        
 [93] reshape2_1.4.4            nlme_3.1-165             
 [95] cachem_1.1.0              stringr_1.5.1            
 [97] parallel_4.3.3            vipor_0.4.7              
 [99] HDO.db_0.99.1             pillar_1.9.0             
[101] grid_4.3.3                vctrs_0.6.5              
[103] promises_1.3.0            BiocSingular_1.18.0      
[105] beachmat_2.18.1           cluster_2.1.6            
[107] beeswarm_0.4.0            evaluate_0.24.0          
[109] cli_3.6.3                 locfit_1.5-9.10          
[111] compiler_4.3.3            rlang_1.1.4              
[113] crayon_1.5.3              labeling_0.4.3           
[115] ps_1.7.6                  getPass_0.2-4            
[117] plyr_1.8.9                fs_1.6.4                 
[119] ggbeeswarm_0.7.2          stringi_1.8.4            
[121] munsell_0.5.1             Biostrings_2.70.3        
[123] lazyeval_0.2.2            GOSemSim_2.28.1          
[125] Matrix_1.6-5              sparseMatrixStats_1.14.0 
[127] bit64_4.0.5               KEGGREST_1.42.0          
[129] statmod_1.5.0             highr_0.11               
[131] igraph_2.0.3              memoise_2.0.1            
[133] bslib_0.7.0               ggtree_3.10.1            
[135] fastmatch_1.1-4           bit_4.0.5                
[137] gson_0.1.0                ape_5.8                  </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
