---
title: "Data Preparation"
subtitle: "00_DataPreparation"
author: "SarahL."
date: "`r format(Sys.time(), '%b %e %Y %H:%M:%S')`"
output: 
  html_document:
    toc: true
    number_sections: true   
    toc_float: true
    code_folding: hide
    df_print: paged
editor_options:
  chunk_output_type: inline
---

```{r setup, include=FALSE, cache=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE)
knitr::opts_chunk$set(cache = FALSE, cache.lazy = FALSE)
knitr::opts_chunk$set(dpi = 300, fig.align = "center")
knitr::opts_chunk$set(out.width = "100%", out.height = "100%")
```


# Data Preparation and Overview

## Setup

### Standard packages

```{r LoadPackages, warning=FALSE}
library(here)
source(here("code", "standard_libraries.R"))
```

### Additional Packages

```{r LoadAddPackages, warning=FALSE}
suppressPackageStartupMessages({
library(DropletUtils)
library(scDblFinder)
})

```

### Set parameter

```{r Setup}
set.seed(123)
bpp <- BiocParallel::MulticoreParam(parallel::detectCores()-1, RNGseed=123)
path <- here::here()
```

## Reports {.tabset}

### CellRanger

View the CellRanger html report output for each sample

- [SynBio028](CellRanger/web_summary_SynBio028.html)\n
- [SynBio049](CellRanger/web_summary_SynBio049.html)\n
- [SynBio050](CellRanger/web_summary_SynBio050.html)\n
- [SynBio074](CellRanger/web_summary_SynBio074.html)\n
- [SynBio076](CellRanger/web_summary_SynBio076.html)\n
- [SynBio077a](CellRanger/web_summary_SynBio077a.html)\n
- [SynBio077b](CellRanger/web_summary_SynBio077b.html)\n
- [SynBio081](CellRanger/web_summary_SynBio081.html)\n
- [SynBio083](CellRanger/web_summary_SynBio083.html)\n
- [SynBio084](CellRanger/web_summary_SynBio084.html)\n
- [SynBio087](CellRanger/web_summary_SynBio087.html)\n
- [SynBio093](CellRanger/web_summary_SynBio093.html)\n
- [SynBio098a](CellRanger/web_summary_SynBio098a.html)\n
- [SynBio098b](CellRanger/web_summary_SynBio098b.html)\n
- [SynBio127](CellRanger/web_summary_SynBio127.html)\n
- [SynBio130](CellRanger/web_summary_SynBio130.html)\n

### CellBender

View the CellBender html report output for each sample

- [SynBio028](CellBender/cellbender_output_SynBio028_10000_w_introns_RL_report.html)\n
- [SynBio049](CellBender/cellbender_output_SynBio049_w_introns_report.html)\n
- [SynBio050](CellBender/cellbender_output_SynBio050_w_introns_report.html)\n
- [SynBio074](CellBender/cellbender_output_SynBio074_w_introns_report.html)\n
- [SynBio076](CellBender/cellbender_output_SynBio076_w_introns_report.html)\n
- [SynBio077a](CellBender/cellbender_output_SynBio077a_w_introns_report.html)\n
- [SynBio077b](CellBender/cellbender_output_SynBio077b_w_introns_RL2_report.html)\n
- [SynBio081](CellBender/cellbender_output_SynBio081_w_introns_RL_report.html)\n
- [SynBio083](CellBender/cellbender_output_SynBio083_w_introns_RL2_report.html)\n
- [SynBio084](CellBender/cellbender_output_SynBio084_w_introns_report.html)\n
- [SynBio087](CellBender/cellbender_output_SynBio087_w_introns_RL2_report.html)\n
- [SynBio093](CellBender/cellbender_output_SynBio093_w_introns_report.html)\n
- [SynBio098a](CellBender/cellbender_output_SynBio098a_w_introns_report.html)\n
- [SynBio098b](CellBender/cellbender_output_SynBio098b_w_introns_report.html)\n
- [SynBio127](CellBender/cellbender_output_SynBio127_w_introns_report.html)\n
- [SynBio130](CellBender/cellbender_output_SynBio130_w_introns_RL2_reporthtml)\n


## Load Data

### Load Cellbender output matrices

Load in the filtered feature barcode matrices from Cellbender of each sample and save them as sce object.

```{r LoadDataset}
#Load Dataset
rawdata_folder <- paste0(path,"/data/cellbender_data_h5/")

#Get file names
filenames <- list.files(rawdata_folder ,recursive = F, full.names = F,pattern = "\\_filtered.h5$")
filepaths <- paste(rawdata_folder,filenames,sep = "")

#Load data as sce object
sce<-read10xCounts(samples = filepaths, sample.names = filenames,col.names=T,type="HDF5")

#Edit Filenames 
colData(sce)$Sample = sub("\\_w_introns.*", "", sub("cellbender_output_", "", as.character(colData(sce)$Sample)))


sce

```

### Load and add Metadata

```{r,LoadMetadata}
#Metadata path
metadata_folder <- paste0(path,"/data/metadata/Metadata_Master.csv")

#Load metadata
metadata_df <-read.csv(metadata_folder, header=TRUE, sep=",")
metadata_df

#Add metadata to sce
colData(sce) <- dplyr::left_join(as.data.frame(colData(sce)),
                                   metadata_df, 
                                   by= c("Sample" = "Sample"),
                                   suffix=c(".x",".y")) %>% 
      #dplyr::select(Sample, Barcode, Diagnosis, sex, Age, Joint.Location, protocol, Pathotype,Krenn total score,) %>% 
      dplyr::select(-one_of("Comments")) %>% #select all except
      DataFrame(row.names=colnames(sce))

#Set Sample names
names(colData(sce))[which(names(colData(sce))=="Sample")]="Orig.Identifier"
names(colData(sce))[which(names(colData(sce))=="ID")]="Sample"

#make col names unique
colnames(sce) <- paste0(sce$Sample, ".", sce$Barcode)



```



### Annotate and Update genes

```{r,UpdateGenes}
#Change Name
names(rowData(sce))[which(names(rowData(sce))=="ID")]="ENSEMBL"

#Annotate genes
AnnoGene <- annotate_genes(data.frame(rowData(sce)), gene_col = "Symbol")

#Add Annotated and Updated genes
rowData(sce) <- AnnoGene

#make row and col names unique
rownames(sce) <- paste0(rowData(sce)$ENSEMBL, ".", rowData(sce)$Symbol)


```



```{r ShowDim}
#Dimensions of count matrix
dim(sce)

```

## Explore dataset

Dimensions of the count matrix


```{r ShowColData}
#Feautures/row data
data.frame(colData(sce))
```

```{r ShowRowData}
#Droplet details / row data
data.frame(rowData(sce))
```

## Exploratory plots

### Histogramm with number of cells {.tabset}

Show the number of cells detected in each sample or joint location before filtering

#### Per Sample

```{r PlotCellsPerSample}
#Histogramm with number of cells per sample
ggplot(colData(sce), aes(x=Sample))+geom_bar()+ coord_flip()+ ggtitle("Number of cells per sample") + theme_bw()
data.frame(as.list(table(colData(sce)$Sample)))

```
#### Per Joint Location

```{r PlotCellsJoint}
#Histogramm with number of cells per sample
ggplot(colData(sce), aes(x=Joint.Location))+geom_bar()+ coord_flip()+ ggtitle("Number of cells per sample") + theme_bw()
data.frame(as.list(table(colData(sce)$Joint.Location)))
```

### Plot number of genes detected per cell

This plot shows cell counts per sample / count occurrence

```{r PlotDetectedPerCell}
#Number of genes detected per cell
#Total UMI for a gene versus the number of times detected
genesPerCell <- colSums(counts(sce) > 0)
plot(density(genesPerCell), xlab="Genes per cell", main="Number of genes detected per cell")

```
### Plot transcript capture efficiency

This plot gives an idea about the sequencing depth and if the sequencing has reached saturation or not. Plotted is the total gene count across all cells (x-axis) vs Proportion of cells the gene is detected in (y-axis) where each dot represents a gene.

```{r PlotCaptureEfficiency}
#transcript_capture_efficiency
#Total UMI for a gene versus the number of times detected
tmpCounts <- counts(sce)

plot(rowSums(tmpCounts),
     rowMeans(tmpCounts > 0),
     log = "x",
     xlab="total number of UMIs",
     ylab="proportion of cells expressing the gene",
     main="Total UMI for a gene vs times detected")

```

## Initial Filtering

Initial filtering to remove unexpressed genes and cells with very low number of counts. The input to scDblFinder should not include empty droplets, and it might be necessary to remove cells with a very low coverage (e.g. <200 reads) to avoid errors. Further quality filtering should be performed downstream of doublet detection.

```{r InitialFiltering}
#Initial filtering before droplet removal
dim_before_filtering <- dim(sce)

#Get only the detected Genes
#Remove Cells with very low counts of less than 150 and genes not expressed 
sce <- sce[rowSums(counts(sce)> 0) > 0, colSums(counts(sce)> 0) > 0] #200


dim_after_filtering <- dim(sce)

cat("NR of Cells Before Initial Filtering: ", dim_before_filtering[2],
    "\nNR of Cells After Initial Filtering: ", dim_after_filtering[2],
    "\nNR of Cells Filtered out: ", dim_before_filtering[2] - dim_after_filtering[2],
    "\nCells Filtered out: [%]", (dim_before_filtering[2] - dim_after_filtering[2])/dim_before_filtering[2]*100,
    
    "\n\nNR of Genes Before Initial Filtering: ", dim_before_filtering[1],
    "\nNR of Genes After Initial Filtering: ", dim_after_filtering[1],
    "\nNR of Genes Filtered out: ", dim_before_filtering[1] - dim_after_filtering[1],
    "\nGenes Filtered out: [%]", (dim_before_filtering[1] - dim_after_filtering[1])/dim_before_filtering[1]*100
    )


```

## Doublet Detection

Doublets are defined as two cells that are sequenced under the same cellular barcode, which happens if they were captured in the same droplet. The scDblFinder method combines the strengths of various doublet detection approaches, training an iterative classifier on the neighborhood of real cells and artificial doublets.

### Detection

```{r DoubletDetection, rows.print=17, warning=FALSE}
#Detection
sce <- scDblFinder::scDblFinder(sce, samples=sce$Sample, BPPARAM = bpp) 

table(sce$scDblFinder.class)
as.data.frame.matrix(table(sce$Sample,sce$scDblFinder.class))

```

### Doublet Detection Plots

#### Singlet/Doublet Histogramm {.tabset}

##### Absolute comparison

```{r DoubletDetectionPlotsAbs}

#Plot singlet/doublet histogramm Absolute comparison / by sample
as.data.frame(colData(sce)) %>%
  dplyr::group_by(Sample, scDblFinder.class) %>%
  dplyr::summarise(Freq=n()) %>%
  ggplot(aes(x=Sample, y=Freq, fill=scDblFinder.class,label=Freq)) + 
    geom_bar(stat="identity") +
    labs(title="Doublet detection results",
    subtitle="By sample",x="",y="Number of Cells") +
    geom_text(size=3, position = position_stack(vjust=0.5))+
    theme(axis.text.x = element_text(angle = 45,hjust=1), axis.line = element_line(colour = "black"), panel.background = element_rect(fill = NA)) 

```

##### Relative comparison

```{r DoubletDetectionPlotsRel}
# Plot singlet/doublet histogramm Relative comparison / by sample
as.data.frame(colData(sce)) %>%
  dplyr::group_by(Sample, scDblFinder.class) %>%
  dplyr::summarise(Freq=n()) %>% 
  ggplot(aes(x=Sample, y=Freq, fill=scDblFinder.class, label=Freq)) +
    geom_bar(stat="identity", position="fill") +
    labs(title="Doublet detection results",
       subtitle="By sample",
       x="Sample",
       y="Number of cells") + 
    geom_text(size=2.5, position = position_fill(vjust=0.5)) +
    theme(axis.text.x = element_text(angle = 45,hjust=1), axis.line = element_line(colour = "black"), panel.background = element_rect(fill = NA)) 
```


#### Singlet/Doublet Scatter

```{r DoubletDetectionPlots}
sce <- addPerCellQC(sce)

#Plot singlet/doublet qcplots
colData(sce) %>% 
    as.data.frame() %>% 
    arrange(scDblFinder.class) %>% 
    ggplot(aes(x = sce$sum, y = sce$detected )) +
      geom_point(aes(colour = scDblFinder.class)) + 
      facet_wrap(vars(Sample))+ 
      labs(title="Total number of detected genes plotted against total number of UMIs",
       x="Total counts",
       y="Detected genes") +
      theme(strip.background=element_rect(fill="white"), panel.background = element_rect(fill = NA),axis.line = element_line(colour = "black")) 


```


### Apply Doublet Removal

```{r DoubletRemoval}

print(paste0("NR of Cells Before Doublet Removal ", dim(sce)[2]))

#Apply Doublet Removal
sce <- sce[ ,sce$scDblFinder.class == "singlet"]

print(paste0("NR of Cells After Doublet Removal ", dim(sce)[2]))

```


## Save the dataset

```{r SaveData}
saveRDS(sce, file =paste0(path,'/output/00_sce_DataPreparation.rds'))
```
