---
title: "Subclustering"
subtitle: "06_SubClustering"
author: "SarahL."
date: "`r format(Sys.time(), '%b %e %Y %H:%M:%S')`"
output: 
  html_document:
    toc: true
    number_sections: true   
    toc_float: true
    code_folding: hide
    df_print: paged
editor_options:
  chunk_output_type: inline
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_chunk$set(warning = FALSE)
knitr::opts_chunk$set(message = FALSE)
knitr::opts_chunk$set(cache = FALSE, cache.lazy = FALSE)
knitr::opts_chunk$set(dpi = 300, fig.align = "center")
knitr::opts_chunk$set(out.width = "100%", out.height = "100%")
```


# Subclustering

## Setup

### Load Packages

```{r LoadPackages, warning=FALSE}
#Load Packages

suppressPackageStartupMessages({
library(BiocParallel)
library(ggplot2)
library(dplyr)
library(scran)
library(scater)
library(bluster)
library(dittoSeq)
library(gridExtra)
})

```

### Set Parameter

```{r Setup}
set.seed(100)
bpp <- BiocParallel::MulticoreParam(parallel::detectCores()-1)
path <- here::here()
```


### Load Data

```{r LoadDataset}
## RA DATASET
sce <- readRDS(file = paste0(path,'/output/05_sce_CelltypeAnnotation.rds'))

```


### Explore Dataset

```{r ShowDim}
#Dimensions of count matrix
dim(sce)
```

https://mperalc.gitlab.io/scRNA-seq_workshop_2021/additional-resources.html## UMAPs coloured by cluster or cell type

### Plot Clustering and Annotation {.tabset}

### Clusters

```{r UMAPCluster}
#cluster
plotReducedDim(sce, "MNN_UMAP", colour_by="cluster", text_by="cluster") +labs(title="UMAP colored by clusters",subtitle = "UMAP of all integrated datasets")

```

### Annotation

```{r UMAPAnnotation}
#cell type
plotReducedDim(sce, "MNN_UMAP", colour_by="celltype", text_by="celltype") +labs(title="UMAP colored by Cell type ",subtitle = "UMAP of all integrated datasets")

```

## Subclustering

```{r SubclusteringFunction}
# Create Subclustering function

SUBCLUSTER <- function(sce.sc, SC_celltypes, clust_param="leiden", k_param=50){

  for ( name in SC_celltypes){
  
    #Subset sce object for ce scelltype
    sce.subset <- sce.sc[,sce.sc$celltype == name]
    
    #Get top HVG for subcluster
    dec <- modelGeneVar(sce.subset, block=sce.subset$Sample, density.weights=FALSE)
    top.hvgs <- getTopHVGs(dec, n=3000)
    rowData(sce.subset)$is_hvg <- rownames(sce.subset) %in% top.hvgs
  
    
    #Batch correction on subcluster
    sce.Batchelor <-batchelor::multiBatchNorm(sce.subset, batch=sce.subset$Sample, subset.row = rownames(sce.subset)[rowData(sce.subset)[["is_hvg"]]], normalize.all=TRUE, BPPARAM = bpp)
    
    bpstart(bpp)
    sce.Batchelor <- batchelor::fastMNN(sce.Batchelor, batch=sce.Batchelor$Sample, prop.k=0.05,  subset.row = rownames(sce.Batchelor)[rowData(sce.Batchelor)[["is_hvg"]]], correct.all=TRUE,BPPARAM = bpp)#merge.order = merge_order,
    bpstop(bpp)
    
    assay(sce.subset, "reconstructed") <- assay(sce.Batchelor, "reconstructed")
    reducedDim(sce.subset, 'MNN') <- reducedDim(sce.Batchelor, 'corrected')
    
    #Recluster the cells 
    clusters <- clusterCells(sce.subset, use.dimred = "MNN", BLUSPARAM = SNNGraphParam(k= k_param,cluster.fun= clust_param))
    
    #Add Subcluster label to sce
    colData(sce.sc)[sce.sc$celltype == name, "subcluster"] <- clusters
  
    #Get UMAP on subset
    sce.subset <- runUMAP(sce.subset, dimred = 'MNN', subset_row=rowData(sce.subset)$is_hvg, min_dist = 0.1)
    umap_for_cell_type <- reducedDim(sce.subset, "UMAP")
    column_name <- paste0("SC_UMAP_", name)
  
    # Add dimension reduction for only selected subcluster cells
    reducedDim(sce.sc, column_name) <- reducedDim(sce.sc, "MNN_UMAP")
    reducedDim(sce.sc, column_name)[,c(1,2)] <- NA
  
    attr(reducedDim(sce.sc, column_name), "scaled:center") <- attr(umap_for_cell_type, "scaled:center")
    reducedDim(sce.sc, column_name)[rownames(umap_for_cell_type),] <- umap_for_cell_type
  }
  
  # transform to factor values
  sce.sc$subcluster_complete <- factor(paste0(sce.sc$celltype, "_", sce.sc$subcluster))
  sce.sc$subcluster <- factor(sce.sc$subcluster)
  
  # Add column for each of the cell type in ColData
  for (name in SC_celltypes){
    colData(sce.sc)[,name] <- colData(sce.sc)[,"subcluster"]
    colData(sce.sc)[sce.sc$celltype != name, name] <- NA
    colData(sce.sc)[, name] <- factor(as.numeric(colData(sce.sc)[, name]))
  }
  
  return(sce.sc)
}
```

```{r RunSubcluster, warning=FALSE}

#Set Celltypes to be subclustered
SC_celltypes <-c("Fibroblast")
#SC_celltypes <-unique(sce$celltype) #All celltypes

sce$celltype <- colData(sce)$ManualAnno_L1

# Run Subcluster Function
sce.sc<-SUBCLUSTER(sce, SC_celltypes, clust_param="leiden", k_param=40)
  
```

## Plot Subclusters

### Plot UMAP of Subclusters  {.tabset}

```{r UMAPSubcluster}
# plot UMAP of the subclusters

for (name in SC_celltypes){
  cat(paste0("#### ", SC_celltypes, " {.tabset} \n"))
  
  cat(paste0("##### Colored by Clustering\n"))
  print(plotReducedDim(sce.sc[,sce.sc$celltype == name], "MNN_UMAP",colour_by = "subcluster", text_by="subcluster") +
          labs(title=paste0("SubCluster ", name),subtitle="Colored by clustering"))
    
  cat(paste0("##### Colored by Sample\n"))
  print(plotReducedDim(sce.sc[,sce.sc$celltype == name], "MNN_UMAP",colour_by = "Sample") +
          labs(title=paste0("SubCluster ", name),subtitle="Colored by Sample"))
}

```

### Plot Recalculated UMAP of Subclusters  {.tabset}

```{r UMAPSubclusterRecalc}
# plot the Recalculated UMAP of the subcluster

for (name in SC_celltypes){
  cat(paste0("#### ", SC_celltypes, " {.tabset} \n"))  
  
  column_name <- paste0("SC_UMAP_", name)
  
  cat(paste0("##### Colored by Clustering\n"))
  print(plotReducedDim(sce.sc[,sce.sc$celltype == name], column_name,colour_by = "subcluster",text_by="subcluster") +
          labs(title=paste0("Recalculated UMAP for Subcluster ", name),subtitle="Colored by clustering"))

  cat(paste0("##### Colored by Sample\n"))
  print(plotReducedDim(sce.sc[,sce.sc$celltype == name], column_name,colour_by = "Sample") +
          labs(title=paste0("Recalculated UMAP for Subcluster ", name),subtitle="Colored by Sample"))
}

```


## Annotate Subclusters 

```{r SetupAssignement}
#Manual Annotation setup
sce.sc$ManualAnno_L2 <- as.integer(sce.sc$celltype)
```

```{r VizGenExFunction}
#Function to visuaise gene Expression for vector of genes

VizGenEx <- function(markr,Ctype){
  for(i in markr){
    
    cat(paste0("#### ", i, "\n"))
    gridExtra::grid.arrange(
      arrangeGrob(plotReducedDim(sce.sc[,sce.sc$celltype == Ctype], dimred = paste0("SC_UMAP_",Ctype), colour_by = i, text_by = "subcluster"), 
                  plotReducedDim(sce.sc[,sce.sc$celltype == Ctype], dimred = paste0("SC_UMAP_",Ctype), colour_by = i, order_by=i ) +scale_colour_gradient(name = i,low = "lightgrey", high = "red"), ncol = 2), 
      plotExpression(sce.sc[,sce.sc$celltype == Ctype], features = i, x = "subcluster"),nrow = 2)
    cat(' \n')
  
  }
}

```

### Fibroblasts

```{r FibroblastsUMAP}
#Plot the UMAp
plotReducedDim(sce.sc[,sce.sc$celltype == "Fibroblast"], "SC_UMAP_Fibroblast" ,colour_by = "subcluster",text_by="subcluster") +
          labs(title=paste0("Recalculated UMAP for Subcluster ", "Fibroblast"),subtitle="Colored by clustering")
```

#### Identify Marker Genes Wilcoxon rank-sum test

```{r IdentifyMarker}
markers_wilcox <- wilcoxauc(sce.sc[,sce.sc$celltype == "Fibroblast"], 'subcluster', assay = 'logcounts')
data.table(markers_wilcox)
```

```{r PlotMarker}
tmp <-top_markers(markers_wilcox, n = 4)

tail(unlist(unname(as.list(tmp))),-4)

dittoDotPlot(sce.sc[,sce.sc$celltype == "Fibroblast"], vars =unique(tail(unlist(unname(as.list(tmp))),-4)) , group.by = "subcluster", min.color="white", max.color="red",scale = FALSE)
```


#### Subcluster cell type Annotation - Level2

```{r AssignementL2, results='asis'}
#Lining Fibroblst
VizGenEx(c("PRG4","CRTAC1", "CLU"),"Fibroblast")  


```

```{r AssignementL2, results='asis'}
#Lining Fibroblst
VizGenEx(c("COL4A1","GAPDH", "POSTN"),"Fibroblast")  


```

```{r AssignementL2}
#Manual Annotation

sce$ManualAnno_L2 <- as.integer(sce$subcluster)
unique(sce$ManualAnno_L2)

sce$ManualAnno_L2[sce$subcluster %in% c(2,3,12)] <- "Fibroblast"

```

#### UMAP Assigned Celltypes Tcell

```{r UMAPAssigned}
plotReducedDim(sce.sc[,sce.sc$celltype == name], column_name,colour_by = "ManualAnno_L2",text_by="ManualAnno_L2") +
          labs(title=paste0("Recalculated UMAP for Subcluster ", name),subtitle="Colored by Celltype")
```


### Tcells



### Macrophages



### Endothelial



```{r AssignementCheck}
#Check if all clusters assigned
unique(sce$ManualAnno_L2)
```

### Plot Assigned Celltypes {.tabset}

##### UMAP All Subcluster

```{r PlotL0}
#Plot manual Annotation
plotReducedDim(sce, dimred="MNN_UMAP", by_exprs_values = "logcounts", colour_by="ManualAnno_L2")+ ggtitle("UMAP Celllabel")
plotReducedDim(sce, dimred="MNN_UMAP", by_exprs_values = "logcounts", colour_by="ManualAnno_L2", text_by="ManualAnno_L2")+ ggtitle("UMAP Celllabel")
```


## Save the dataset

```{r SaveData}
saveRDS(sce, file =paste0(path,'/output/06_sce_SubClustering.rds'))
```
