---
title: "Differential Expression Analysis Pseudobulk"
subtitle: "07_DifferentialExpression_Pseudobulk"
author: "SarahL."
date: "`r format(Sys.time(), '%b %e %Y %H:%M:%S')`"
output: 
  html_document:
    toc: true
    number_sections: true   
    toc_float: true
    code_folding: hide
    df_print: paged
editor_options:
  chunk_output_type: inline
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE)
knitr::opts_chunk$set(cache = FALSE, cache.lazy = FALSE)
knitr::opts_chunk$set(dpi = 300, fig.align = "center")
knitr::opts_chunk$set(out.width = "100%", out.height = "100%")
```


# Differential Expression Analysis

## Setup

### Standard packages

```{r LoadPackages, warning=FALSE}
library(here)
source(here("code", "standard_libraries.R"))
```

### Additional Packages

```{r LoadPackages, warning=FALSE}
#Load Packages

suppressPackageStartupMessages({
library(muscat)
library(limma)
library(EnhancedVolcano)
})

```

### Set Parameter

```{r Setup}
set.seed(100)
bpp <- BiocParallel::MulticoreParam(parallel::detectCores()-1, RNGseed=100)
path <- here::here()
```


### Load Data

```{r LoadDataset}
## RA DATASET
sce <- readRDS(file = paste0(path,'/output/05_sce_CelltypeAnnotation.rds'))

#Select only Fibroblast and Myeloid
selected_celltype <- c("Fibroblast","Myeloid")
sce <- sce[,sce$celltype == selected_celltype]

```

### Explore Dataset

```{r ShowDim}
#Dimensions of count matrix
dim(sce)
```

https://mperalc.gitlab.io/scRNA-seq_workshop_2021/additional-resources.html## UMAPs coloured by cluster or cell type

### Plot Clustering and Annotation {.tabset}

#### Annotation

```{r UMAPAnnotation}
#cell type
plotReducedDim(sce, "MNN_UMAP", colour_by="celltype", text_by="celltype") +labs(title="UMAP colored by Cell type ",subtitle = "UMAP of all integrated datasets")

```

#### Clusters

```{r UMAPCluster}
#cluster
plotReducedDim(sce, "MNN_UMAP", colour_by="cluster", text_by="cluster") +labs(title="UMAP colored by clusters",subtitle = "UMAP of all integrated datasets")

```

## Differential Expression Pseudobulk

### Setup

#### Sample preparation

```{r DE}
# create SCE
sce$id <- paste0(sce$Joint.Location, sce$Sample)
(sce <- prepSCE(sce, 
    kid = "label", # Celllabels
    gid = "Joint.Location",  # Condition
    sid = "id",   # Samplenames
    drop = TRUE))  # drop all other colData columns
```

#### Aggregated Data Overview

```{r DE, rows.print=20}
# nr of cells per cluster-sample
as.data.frame.matrix(t(table(sce$cluster_id, sce$sample_id)))
```


#### Creating pseudo-bulk

```{r DE}
# Aggregate Cells
pb <- aggregateData(sce,
    assay = "counts", fun = "sum",
    by = c("cluster_id", "sample_id"))
# one sheet per subpopulation
assayNames(pb)
t(head(assay(pb)))
```


```{r DE}
#MDS plot
pbMDS(pb)
```

```{r DE}
# construct design & contrast matrix
ei <- metadata(sce)$experiment_info
mm <- model.matrix(~ 0 + ei$group_id)
dimnames(mm) <- list(ei$sample_id, levels(ei$group_id))
x <- c("Knee-Wrist","Knee-MCP","Wrist-MCP")
contrast <- makeContrasts(contrasts=x, levels = mm)
contrast

```

### Run DS analysis

```{r DE}
# run DS analysis

res_edgeR<-pbDS(pb, design = mm, contrast = contrast, method = c("edgeR"))
res_DESeq2<-pbDS(pb, design = mm, contrast = contrast, method = c("DESeq2"))
res_limmatrend<-pbDS(pb, design = mm, contrast = contrast, method = c("limma-trend"))
res_limmavoom<-pbDS(pb, design = mm, contrast = contrast, method = c("limma-voom"))

```

## Results for Differential Expression

### View results for first comparison (Knee-Wrist)

```{r DE}
# View results for first comparison (Knee-Wrist)
for (method in c("edgeR","DESeq2","limmatrend","limmavoom")){
    
  #Select correct DEG table
  tbl <- res_edgeR$table[["Knee-Wrist"]]
    
  #Take the significant entries of the table
  tbl_fil <- lapply(tbl, function(u) {
    u <- dplyr::filter(u, p_adj.loc < 0.05, abs(logFC) > 0.1)
    dplyr::arrange(u, p_adj.loc)
  })
    
  for (C_type in selected_celltype){
    
    #Select correct celltype
    tbl_fil_CT <- tbl_fil$Fibroblast[1]
    
    dd <- data.frame()
    dd$edgeR <-integer()
    
    if 
    #empty dataframe of ctype
    #append colum of the method
    #append row of gene and add 0 to already existing colums if gene not in table already
    #entries either 1 or 0 if gene is DEG
    
    
    
  }
  
}


# access results table for 1st comparison
tbl <- res_edgeR$table[["Knee-Wrist"]]
# show clusters/labels
names(tbl)
```


```{r DE}
# filter FDR < 5%, abs(logFC) > 1 & sort by adj. p-value
tbl_fil <- lapply(tbl, function(u) {
  u <- dplyr::filter(u, p_adj.loc < 0.05, abs(logFC) > 0.1)
  dplyr::arrange(u, p_adj.loc)
})

# nb. of DS genes & % of total by cluster
n_de <- vapply(tbl_fil, nrow, numeric(1))
p_de <- format(n_de / nrow(sce) * 100, digits = 3)
data.frame("#DS" = n_de, "%DS" = p_de, check.names = FALSE)
```

## Upset plot of DEGS

```{r SaveData}
plot
```


## Save the dataset

```{r SaveData}

saveRDS(sce, file =paste0(path,''))

```
