---
title: "Differential Expression Analysis"
subtitle: "07_DifferentialExpression_EdgeR"
author: "SarahL."
date: "`r format(Sys.time(), '%b %e %Y %H:%M:%S')`"
output: 
  html_document:
    toc: true
    number_sections: true   
    toc_float: true
    code_folding: hide
    df_print: paged
editor_options:
  chunk_output_type: inline
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_chunk$set(warning = FALSE)
knitr::opts_chunk$set(message = FALSE)
knitr::opts_chunk$set(cache = FALSE, cache.lazy = FALSE)
knitr::opts_chunk$set(dpi = 300, fig.align = "center")
knitr::opts_chunk$set(out.width = "100%", out.height = "100%")
```


# Differential Expression Analysis

## Setup

### Load Packages

```{r LoadPackages, warning=FALSE}
#Load Packages

suppressPackageStartupMessages({
library(BiocParallel)
library(ggplot2)
library(dplyr)
library(scran)
library(edgeR)
library(EnhancedVolcano)
})

```

### Set Parameter

```{r Setup}
set.seed(100)
bpp <- BiocParallel::MulticoreParam(parallel::detectCores()-1, RNGseed=100)
path <- here::here()
```


### Load Data

```{r LoadDataset}
## RA DATASET
sce <- readRDS(file = paste0(path,'/output/05_sce_CelltypeAnnotation.rds'))

```


### Explore Dataset

```{r ShowDim}
#Dimensions of count matrix
dim(sce)
```

https://mperalc.gitlab.io/scRNA-seq_workshop_2021/additional-resources.html## UMAPs coloured by cluster or cell type

### Plot Clustering and Annotation {.tabset}

### Annotation

```{r UMAPAnnotation}
#cell type
plotReducedDim(sce, "MNN_UMAP", colour_by="celltype", text_by="celltype") +labs(title="UMAP colored by Cell type ",subtitle = "UMAP of all integrated datasets")

```

### Clusters

```{r UMAPCluster}
#cluster
plotReducedDim(sce, "MNN_UMAP", colour_by="cluster", text_by="cluster") +labs(title="UMAP colored by clusters",subtitle = "UMAP of all integrated datasets")

```

## Differential Expression

### Set Celltype Annotation to use

```{r DE}
#sce$label <-
# Nr of cells in celltypes per Sample
as.data.frame.matrix(table(sce$label, sce$Sample))
```



### Creating pseudo-bulk

```{r DE}
# Aggregate Cells
sce.summ <- aggregateAcrossCells(sce, id=colData(sce)[,c("label", "Sample")])
sce.summ
```


```{r DE}
# filter pseudo-bulk object
Ctype <- "Fibroblast"
current <- sce.summ[,Ctype==sce.summ$label]

# Creating up a DGEList object for use in edgeR:
y <- DGEList(counts(current), samples=colData(current))
y
```

```{r DE}
# filter pseudo-bulk object

discarded <- current$ncells < 10
summary(discarded)
y <- y[,!discarded]

keep <- filterByExpr(y, group=current$	
Joint.Location)
summary(keep)
y <- y[keep,]

```

```{r DE}
# normalisation correct for composition biases by computing normalization factors with the trimmed mean of M-values method
y <- calcNormFactors(y)
y$samples
```

```{r DE}
#mean-difference (MD) plot for each normalized pseudo-bulk profile 
par(mfrow=c(2,3))
for (i in seq_len(ncol(y))) {
    plotMD(y, column=i)
    abline(h = 0, col = "blue", lwd = 2)
}
```
```{r DE}
#mean-difference (MD) plot for each normalized pseudo-bulk profile 
colors <- c("blue", "red","green")
plotMDS(cpm(y, log=TRUE), col=colors[factor(y$samples$Joint.Location)])
```

### Performing Differential Expression


```{r DE}
# set up the design matrix

design <- model.matrix(~factor(Joint.Location), y$sample)
design

```

```{r DE}
# set up the design matrix

y <- estimateDisp(y, design)
summary(y$trended.dispersion)
plotBCV(y)
```

```{r DE}
fit <- glmQLFit(y, design, robust=TRUE)
summary(fit$var.prior)
summary(fit$df.prior)
plotQLDisp(fit)
```
```{r DE}
res <- glmQLFTest(fit, coef=ncol(design)-1)
summary(decideTests(res))
topTags(res)
```

```{r DE}
res <- glmQLFTest(fit, coef=ncol(design))
summary(decideTests(res))
topTags(res)
```

## DE looping across labels

### DE Knee vs Wrist

```{r DE}
# Pre-processing for DE Knee <-> Wrist

sce.summ.filt <- sce.summ[,sce.summ$Joint.Location!= "MCP"]
sce.summ.filt <- sce.summ.filt[,sce.summ.filt$ncells >= 10]

#sce.summ.filt$Joint.Location<-factor(sce.summ.filt$Joint.Location)

de.results <- pseudoBulkDGE(sce.summ.filt, 
    sample=sce.summ.filt$Sample,
    label=sce.summ.filt$label,
    design=~Joint.Location,
    coef="Joint.LocationWrist",
    condition=sce.summ.filt$Joint.Location 
)

```


```{r SaveData}
#show failed labels
metadata(de.results)$failed
```


```{r SaveData}
#  DE results for Fibroblast
cur.results <- de.results[["Fibroblast"]]
cur.results[order(cur.results$PValue),]
```


```{r SaveData,  out.width = '30%',fig.height = 7}
#  DE results for Fibroblast
EnhancedVolcano(cur.results,lab = rownames(cur.results),x = 'logFC', y = 'PValue', title = paste("Differential expression", " Knee ", "vs", " Wrist ", sep=' '),subtitle = "Fibroblast cluster (pseudobulk)",colAlpha = 1)
```


```{r SaveData}
is.de <- decideTestsPerLabel(de.results, threshold=0.05)
summarizeTestsPerLabel(is.de)
```

```{r SaveData}
# Upregulated across most cell types.
up.de <- is.de > 0 & !is.na(is.de)
head(sort(rowMeans(up.de), decreasing=TRUE), 10)
```


```{r SaveData}
# Downregulated across cell types.
down.de <- is.de < 0 & !is.na(is.de)
head(sort(rowMeans(down.de), decreasing=TRUE), 10)
```





### identify label-specific DE

```{r SaveData}
de.specific <- pseudoBulkSpecific(sce.summ.filt,
    label=sce.summ.filt$label,
    design=~Joint.Location,
    coef="Joint.LocationWrist",
    condition=sce.summ.filt$Joint.Location 
)
```


```{r SaveData}
#expression of one gene across all celltypes
sizeFactors(sce.summ.filt) <- NULL
plotExpression(logNormCounts(sce.summ.filt),
    features="TPST1",
    x="Joint.Location", colour_by="Joint.Location",
    other_fields="label") +
    facet_wrap(~label)

```

### identify genes that are significant in label of interest yet not DE in any other label

```{r SaveData}
# Finding all genes that are not remotely DE in all other labels.
remotely.de <- decideTestsPerLabel(de.results, threshold=0.5)
not.de <- remotely.de==0 | is.na(remotely.de)
not.de.other <- rowMeans(not.de[,colnames(not.de)!="Fibroblast"])==1

# Intersecting with genes that are DE in Fibroblast
unique.degs <- is.de[,"Fibroblast"]!=0 & not.de.other
unique.degs <- names(which(unique.degs))

# Inspecting the results.
de.Fibroblast <- de.results$Fibroblast
de.Fibroblast <- de.Fibroblast[unique.degs,]
de.Fibroblast <- de.Fibroblast[order(de.Fibroblast$PValue),]
de.Fibroblast
```

```{r SaveData}
#expression of one gene across all celltypes
sizeFactors(sce.summ.filt) <- NULL
plotExpression(logNormCounts(sce.summ.filt),
    features="HOXD11",
    x="Joint.Location", colour_by="Joint.Location",
    other_fields="label") +
    facet_wrap(~label)

```


## Save the dataset

```{r SaveData}
saveRDS(sce, file =paste0(path,'/output/07_sce_DifferentialExpression.rds'))
```
