---
title: "Differential Expression Analysis"
subtitle: "07_DifferentialExpression_muscat"
author: "SarahL."
date: "`r format(Sys.time(), '%b %e %Y %H:%M:%S')`"
output: 
  html_document:
    toc: true
    number_sections: true   
    toc_float: true
    code_folding: hide
    df_print: paged
editor_options:
  chunk_output_type: inline
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_chunk$set(warning = FALSE)
knitr::opts_chunk$set(message = FALSE)
knitr::opts_chunk$set(cache = FALSE, cache.lazy = FALSE)
knitr::opts_chunk$set(dpi = 300, fig.align = "center")
knitr::opts_chunk$set(out.width = "100%", out.height = "100%")
```


# Differential Expression Analysis

## Setup

### Load Packages

```{r LoadPackages, warning=FALSE}
#Load Packages

suppressPackageStartupMessages({
library(BiocParallel)
library(ggplot2)
library(dplyr)
library(scran)
library(muscat)
library(limma)
library(EnhancedVolcano)
})

```

### Set Parameter

```{r Setup}
set.seed(100)
bpp <- BiocParallel::MulticoreParam(parallel::detectCores()-1, RNGseed=100)
path <- here::here()
```


### Load Data

```{r LoadDataset}
## RA DATASET
sce <- readRDS(file = paste0(path,'/output/05_sce_CelltypeAnnotation.rds'))

```


### Explore Dataset

```{r ShowDim}
#Dimensions of count matrix
dim(sce)
```

https://mperalc.gitlab.io/scRNA-seq_workshop_2021/additional-resources.html## UMAPs coloured by cluster or cell type

### Plot Clustering and Annotation {.tabset}

#### Annotation

```{r UMAPAnnotation}
#cell type
plotReducedDim(sce, "MNN_UMAP", colour_by="celltype", text_by="celltype") +labs(title="UMAP colored by Cell type ",subtitle = "UMAP of all integrated datasets")

```

#### Clusters

```{r UMAPCluster}
#cluster
plotReducedDim(sce, "MNN_UMAP", colour_by="cluster", text_by="cluster") +labs(title="UMAP colored by clusters",subtitle = "UMAP of all integrated datasets")

```

## Differential Expression

### Set Celltype Annotation to use

```{r DE}
#sce$label <-
# Nr of cells in celltypes per Sample
as.data.frame.matrix(table(sce$label, sce$Sample))
```

### Sample preparation

```{r DE}
#prepSCE
sce$id <- paste0(sce$Joint.Location, sce$Sample)
(sce <- prepSCE(sce, 
    kid = "label", # Celllabels
    gid = "Joint.Location",  # Condition
    sid = "id",   # Samplenames
    drop = TRUE))  # drop all other colData columns
```

```{r DE, rows.print=20}
# nb. of cells per cluster-sample
as.data.frame.matrix(t(table(sce$cluster_id, sce$sample_id)))
```


### Creating pseudo-bulk

```{r DE}
# Aggregate Cells
pb <- aggregateData(sce,
    assay = "counts", fun = "sum",
    by = c("cluster_id", "sample_id"))
# one sheet per subpopulation
assayNames(pb)
t(head(assay(pb)))
```


```{r DE}
#MDS plot
pbMDS(pb)
```

```{r DE}
# construct design & contrast matrix
ei <- metadata(sce)$experiment_info
mm <- model.matrix(~ 0 + ei$group_id)
dimnames(mm) <- list(ei$sample_id, levels(ei$group_id))
x <- c("Knee-Wrist","Knee-MCP","Wrist-MCP")
contrast <- makeContrasts(contrasts=x, levels = mm)
contrast

```

### Run DS analysis

```{r DE}
# run DS analysis
res<-pbDS(pb, design = mm, contrast = contrast, method = c("DESeq2"))

```
## Results for Differential Expression

### View results for first comparison (Knee-Wrist)

```{r DE}
# View results for first comparison (Knee-Wrist)

# access results table for 1st comparison
tbl <- res$table[["Knee-Wrist"]]
# show clusters/labels
names(tbl)
```


```{r DE}
# filter FDR < 5%, abs(logFC) > 1 & sort by adj. p-value
tbl_fil <- lapply(tbl, function(u) {
  u <- dplyr::filter(u, p_adj.loc < 0.05, abs(logFC) > 1)
  dplyr::arrange(u, p_adj.loc)
})

# nb. of DS genes & % of total by cluster
n_de <- vapply(tbl_fil, nrow, numeric(1))
p_de <- format(n_de / nrow(sce) * 100, digits = 3)
data.frame("#DS" = n_de, "%DS" = p_de, check.names = FALSE)
```


```{r DE}
# view top hits in each cluster
topn <- bind_rows(lapply(tbl_fil, top_n, -40, p_adj.loc))
format(topn[, -ncol(topn)], digits = 2)
```

```{r DE}
# top-5 DS genes per cluster
pbHeatmap(sce, tbl_fil, top_n = 5,sort_by="logFC")
```

#### View results for individual celltypes


```{r DE, fig.height=10, results='asis'}
for (Ctype in names(n_de[n_de!=0])){
  
  cat(paste0("#### ", Ctype, "\n"))
  
  TableCtype <- tbl[[Ctype]]
  print(head(dplyr::arrange(format(TableCtype[, -ncol(TableCtype)], digits = 2), as.numeric(p_adj.loc)),n = -100L))
  
  
  print(
  EnhancedVolcano(TableCtype, lab = TableCtype$gene,
                x = 'logFC', y = 'p_adj.loc', 
                pCutoff = 0.05, FCcutoff = 1,
                title = paste("Differential expression", " Knee ", "vs", " Wrist ", sep=' '),
                subtitle = paste0(Ctype," cluster (pseudobulk)"),
                pointSize = 2.0, colAlpha = 1,
                legendPosition = 'right')
  )
  
  
  cat(' \n')
  
  
  
}



```

```{r DE }

plotExpression(sce[, sce$cluster_id == "Fibroblast"],
  features = tbl_fil$`Fibroblast`$gene[seq_len(6)],
  x = "sample_id", colour_by = "group_id", ncol = 3) +
  guides(fill = guide_legend(override.aes = list(size = 5, alpha = 1))) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

pbHeatmap(sce, res, k = "Fibroblast",sort_by = "logFC")

```


### View results for second comparison (Knee-MCP)

```{r DE}
# View results for first comparison (Knee-Wrist)

# access results table for 1st comparison
tbl <- res$table[["Knee-MCP"]]
# show clusters/labels
names(tbl)
```


```{r DE}
# filter FDR < 5%, abs(logFC) > 1 & sort by adj. p-value
tbl_fil <- lapply(tbl, function(u) {
  u <- dplyr::filter(u, p_adj.loc < 0.05, abs(logFC) > 1)
  dplyr::arrange(u, p_adj.loc)
})

# nb. of DS genes & % of total by cluster
n_de <- vapply(tbl_fil, nrow, numeric(1))
p_de <- format(n_de / nrow(sce) * 100, digits = 3)
data.frame("#DS" = n_de, "%DS" = p_de, check.names = FALSE)
```


```{r DE,rows.print=25}
# view top hits in each cluster
topn <- bind_rows(lapply(tbl_fil, top_n, 200, -p_adj.loc))
format(topn[, -ncol(topn)], digits = 2)
```


```{r DE}
# top-5 DS genes per cluster
pbHeatmap(sce, tbl_fil, top_n = 25,sort_by="logFC")
```


#### View results for individual celltypes

```{r DE, fig.height=10, results='asis'}
for (Ctype in names(n_de[n_de!=0])){
  
  cat(paste0("#### ", Ctype, "\n"))
  
  TableCtype <- tbl[[Ctype]]
  print(head(dplyr::arrange(format(TableCtype[, -ncol(TableCtype)], digits = 2), as.numeric(p_adj.loc)),n = -100L))
  
  
  print(
  EnhancedVolcano(TableCtype, lab = TableCtype$gene,
                x = 'logFC', y = 'p_adj.loc', 
                pCutoff = 0.05, FCcutoff = 1,
                title = paste("Differential expression", " Knee ", "vs", " MCP ", sep=' '),
                subtitle = paste0(Ctype," cluster (pseudobulk)"),
                pointSize = 2.0, colAlpha = 1,
                legendPosition = 'right')
  )
  cat(' \n')
  
}


```



### View results for third comparison (Wrist-MCP)

```{r DE}
# View results for first comparison (Wrist-MCP)

# access results table for 1st comparison
tbl <- res$table[["Wrist-MCP"]]
# show clusters/labels
names(tbl)
```


```{r DE}
# filter FDR < 5%, abs(logFC) > 1 & sort by adj. p-value
tbl_fil <- lapply(tbl, function(u) {
  u <- dplyr::filter(u, p_adj.loc < 0.05, abs(logFC) > 1)
  dplyr::arrange(u, p_adj.loc)
})

# nb. of DS genes & % of total by cluster
n_de <- vapply(tbl_fil, nrow, numeric(1))
p_de <- format(n_de / nrow(sce) * 100, digits = 3)
data.frame("#DS" = n_de, "%DS" = p_de, check.names = FALSE)
```


```{r DE}
# view top hits in each cluster
topn <- bind_rows(lapply(tbl_fil, top_n, -200, p_adj.loc))
format(topn[, -ncol(topn)], digits = 2)
```


```{r DE}
# top-5 DS genes per cluster
pbHeatmap(sce, tbl_fil, top_n = 5,sort_by="logFC")
```

#### View results for individual celltypes

```{r DE, fig.height=10, results='asis'}
for (Ctype in names(n_de[n_de!=0])){
  
  cat(paste0("#### ", Ctype, "\n"))
  
  TableCtype <- tbl[[Ctype]]
  print(head(dplyr::arrange(format(TableCtype[, -ncol(TableCtype)], digits = 2), as.numeric(p_adj.loc)),n = -100L))
  
  
  print(
  EnhancedVolcano(TableCtype, lab = TableCtype$gene,
                x = 'logFC', y = 'p_adj.loc', 
                pCutoff = 0.05, FCcutoff = 1,
                title = paste("Differential expression", " Wrist ", "vs", " MCP ", sep=' '),
                subtitle = paste0(Ctype," cluster (pseudobulk)"),
                pointSize = 2.0, colAlpha = 1,
                legendPosition = 'right')
  )
  cat(' \n')
  
}



```




## Save the dataset

```{r SaveData}
#resDS(sce, res, bind = "row", frq = TRUE)

saveRDS(sce, file =paste0(path,'/output/07_sce_DifferentialExpression.rds'))
```
