---
title: "Differential Abundance Analysis"
subtitle: "08_DifferentialAbundance_miloR"
author: "SarahL."
date: "`r format(Sys.time(), '%b %e %Y %H:%M:%S')`"
output: 
  html_document:
    toc: true
    number_sections: true   
    toc_float: true
    code_folding: hide
    df_print: paged
editor_options:
  chunk_output_type: inline
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_chunk$set(warning = FALSE)
knitr::opts_chunk$set(message = FALSE)
knitr::opts_chunk$set(cache = FALSE, cache.lazy = FALSE)
knitr::opts_chunk$set(dpi = 300, fig.align = "center")
knitr::opts_chunk$set(out.width = "100%", out.height = "100%")
```


# Differential Expression Analysis

## Setup

### Load Packages

```{r LoadPackages, warning=FALSE}
#Load Packages

suppressPackageStartupMessages({
library(BiocParallel)
library(ggplot2)
library(dplyr)
library(scran)
library(scater)
library(miloR)
library(patchwork)
})

```

### Set Parameter

```{r Setup}
set.seed(100)
bpp <- BiocParallel::MulticoreParam(parallel::detectCores()-1, RNGseed=100)
path <- here::here()
```


### Load Data

```{r LoadDataset}
## RA DATASET
sce <- readRDS(file = paste0(path,'/output/03_sce_Integration_Batchelor.rds'))

```


### Explore Dataset

```{r ShowDim}
#Dimensions of count matrix
dim(sce)
```

https://mperalc.gitlab.io/scRNA-seq_workshop_2021/additional-resources.html## UMAPs coloured by cluster or cell type

### Plot Clustering and Annotation {.tabset}

#### Annotation

```{r UMAPAnnotation}
#cell type
plotReducedDim(sce, "MNN_UMAP", colour_by="celltype", text_by="celltype") +labs(title="UMAP colored by Annotation ",subtitle = "UMAP of all integrated datasets")

```

#### Joint Location

```{r UMAPCluster}
#cluster
plotReducedDim(sce, "MNN_UMAP", colour_by="Joint.Location") +labs(title="UMAP colored by Joint Location",subtitle = "UMAP of all integrated datasets")

```

### Frequencies of celltypes in clusters per Joint Location {.tabset}

```{r ClusterInfo}
ClusterInfo <- as.data.frame(colData(sce)) %>% 
  group_by(celltype, Joint.Location) %>%
  summarise(cells = n(),.groups = 'drop') 
```

#### Absolute comparison

```{r PlotAbsoluteClustering}

ggplot(data=ClusterInfo, aes(x=Joint.Location,y=cells, fill=celltype)) +
  geom_bar(stat="identity", position="stack") +
  labs(title="Absolute comparison nr Cells in Celltype",subtitle="Per Joint",x="",y="Number of cells")+
  theme(axis.text.x = element_text(angle = 45,hjust=1), axis.ticks.x=element_blank())

```

#### Relative comparison

```{r PlotRelativeClustering}

ggplot(data=ClusterInfo,aes(x=Joint.Location, y=cells, fill=celltype)) +
  geom_bar(stat="identity", position="fill") +
  labs(title="Relative comparison nr Cells in celltype",subtitle="Per Joint",x="",y="Number of cells")+
  theme(axis.text.x = element_text(angle = 45,hjust=1), axis.ticks.x=element_blank())


```



## Differential Abundance with miloR


### Sample preparation

```{r }
#Create milo object

sce.milo <- Milo(sce)

```

```{r }
#Decide on d parameter

elbow <- ceiling(as.numeric(intrinsicDimension::maxLikGlobalDimEst(as.matrix(reducedDim(sce, "PCA")), k=10)))

# Extract variance explained
pca_pct_variance <- data.frame(variance = attr(reducedDim(sce, "PCA"), "percentVar"))
pca_pct_variance$PC <- 1:nrow(pca_pct_variance)

ggplot(pca_pct_variance,aes(PC, variance)) +geom_line(color="grey") +geom_point() +labs(y = "Variance explained (%)")+ geom_vline(xintercept = elbow, color="red")

```


### Construct KNN graph


```{r }
#construct KNN graph

sce.milo <- buildGraph(sce.milo, k = 40, d = 20, reduced.dim = "MNN_reduced")

```


```{r }
#Defining representative neighbourhoods on the KNN graph
sce.milo <- makeNhoods(sce.milo, prop = 0.05, k = 40, d = 20, refined = TRUE, reduced_dims = "MNN_reduced",refinement_scheme="graph")

# To evaluate whether the value of k used for graph building was appropriate. We can check this out using the plotNhoodSizeHist function. As a rule of thumb we want to have an average neighbourhood size over 5 x N_samples. If the mean is lower, or if the distribution is

plotNhoodSizeHist(sce.milo)

```


```{r }
# Counting cells in neighbourhoods

sce.milo <- countCells(sce.milo, meta.data = data.frame(colData(sce.milo)), sample="Sample")
head(nhoodCounts(sce.milo))

```

### Defining experimental design

```{r }
# Defining experimental design
design <- data.frame(colData(sce.milo))[,c("Sample", "Joint.Location")]

## Convert batch info from integer to factor
design$Joint.Location <- as.factor(design$Joint.Location) 
design <- distinct(design)
rownames(design) <- design$Sample

## Reorder rownames to match columns of nhoodCounts(milo)
design <- design[colnames(nhoodCounts(sce.milo)), , drop=FALSE]

design
```

```{r }
# Set model matrix
mm <- model.matrix(~ 0 + Joint.Location, data=design)
#dimnames(mm)[2] <- list(levels(design$Joint.Location))
mm
# Set contrasts
x <- c("Joint.LocationKnee-Joint.LocationWrist","Joint.LocationKnee-Joint.LocationMCP","Joint.LocationWrist-Joint.LocationMCP")
constrast <- makeContrasts(contrasts=x, levels=mm)
constrast
```


### Computing neighbourhood connectivity & testing

```{r }
# Computing neighbourhood connectivity
sce.milo <- calcNhoodDistance(sce.milo, d=20, reduced.dim = "MNN_reduced")

```

```{r }
# Testing
res <- testNhoods(sce.milo, design = ~ 0 + Joint.Location, design.df = design, model.contrasts = "Joint.LocationKnee-Joint.LocationWrist", reduced.dim = "MNN_reduced", fdr.weighting="graph-overlap") # fdr.weighting="graph-overlap"

```


### Inspecting DA results


```{r }
res %>%
  arrange(SpatialFDR) %>%
  head()

table(res$SpatialFDR < 0.1)
```

#### Distribution of uncorrected P values

We expect a peak to very low extreme and a tail towards 1

```{r }
ggplot(res, aes(PValue)) + geom_histogram(bins=50)
```

#### Volcano plot of neighbourhoods

Check to see whether neighborhoods were there is a significant difference where detected in cell abundances between sample groups.

```{r }
ggplot(res, aes(logFC, -log10(SpatialFDR))) + geom_point() +geom_hline(yintercept = 1, color="red") 
```

#### Abstracted graph of neighbourhoods

```{r }
sce.milo <- buildNhoodGraph(sce.milo)

## Plot single-cell UMAP
umap_pl <- plotReducedDim(sce.milo, dimred = "MNN_UMAP_reduced3", 
                          text_size = 3, point_size=0.5) +
  guides(fill="none")

## Plot neighbourhood graph
nh_graph_pl <- plotNhoodGraphDA(sce.milo, res, layout="MNN_UMAP_reduced3",alpha=0.1) 
  
umap_pl + nh_graph_pl +
  plot_layout(guides="auto")

```
#### Assign cell labels to each neighbourhood 

```{r }
res <- annotateNhoods(sce.milo, res, coldata_col = "label")
res
```

Plot to see what the distribution of neighbourhoods is like within our labels. We could take this as a measure of how similar our clustering is with the KNN neighborhood as defined by Milo

```{r }
ggplot(res, aes(label_fraction)) + geom_histogram(bins=50)
```


```{r }
# Define a threshold for label_fraction to exclude neighbourhoods that are a mix of cell types
res$label <- ifelse(res$label_fraction < 0.8, "Mixed", res$label)
```

```{r }
# Visualise the distribution of DA Fold Changes in different labels.
plotDAbeeswarm(res, group.by = "label")
```

## Save the dataset

```{r SaveData}
#resDS(sce, res, bind = "row", frq = TRUE)

saveRDS(sce.milo, file =paste0(path,'/output/08_DifferentialAbundance_miloR.rds'))
```
